tcms/static/js/testrun_actions.js

```diff
@@ -244,63 +257,44 @@ Nitrate.TestRuns.AssignCase.on_load = function() {
   });
 };
 
+function updateExecutionStatus(updateForm, executionId, caseId, executionStatusPk) {
 
-function updateRunStatus(object_pk, value, callback) {
-  jQ.ajax({
-    'url': '/runs/case-run-update-status/',
-    'type': 'POST',
-    'data': {'object_pk': object_pk, 'status_id': value },
-    'success': function (data, textStatus, jqXHR) {
-      callback();
-    },
-    'error': function (jqXHR, textStatus, errorThrown) {
-      json_failure(jqXHR);
-    }
-  });
-}
+  const container = updateForm.parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+  const comment = updateForm.find('textarea[name=comment]').val()
+  if (comment) {
+    addComment(executionId, caseId, comment)
+  }
```

Comment:
The text from the textarea field is not removed after a comment is added. For the moment the `addComment()` function is used only in 2 places - here and on line 81 (where only comment is added via the Submit button. You can refactor the function to accept the container element instead of the text and only post the comment if the text is not empty. Then clear the text from the container.